pgjdbc/src/test/java/org/postgresql/test/jdbc2/PreparedStatementTest.java

```diff
@@ -1598,4 +1598,131 @@ public void testNoParametersNPE() throws SQLException {
+        + "INSERT INTO inttable VALUES (?)");
+    pstmt.setInt(1, 1);
+    int count = pstmt.executeUpdate();
+    pstmt.close();
+    assertEquals("Expected update count to be 1", 1, count);
+    //With two SET statements
+    pstmt = con.prepareStatement("SET search_path = 'public';SET search_path = 'public';"
+        + "INSERT INTO inttable VALUES (?)");
+    pstmt.setInt(1, 1);
+    count = pstmt.executeUpdate();
+    pstmt.close();
+    assertEquals("Expected update count to be 1", 1, count);
+
+    //UPDATE
+    pstmt = con.prepareStatement("SET search_path = 'public';"
+        + "UPDATE inttable SET a=?");
+    pstmt.setInt(1, 1);
+    count = pstmt.executeUpdate();
+    pstmt.close();
+    assertEquals("Expected update count to be 2", 2, count);
+    //With two SET statements
+    pstmt = con.prepareStatement("SET search_path = 'public';SET search_path = 'public';"
+        + "UPDATE inttable SET a=?");
+    pstmt.setInt(1, 1);
+    count = pstmt.executeUpdate();
+    pstmt.close();
+    assertEquals("Expected update count to be 2", 2, count);
+
+    //DELETE
+    pstmt = con.prepareStatement("SET search_path = 'public';"
+        + "DELETE FROM inttable WHERE a=?");
+    pstmt.setInt(1, 1);
+    count = pstmt.executeUpdate();
+    pstmt.close();
+    assertEquals("Expected update count to be 2", 2, count);
+    //With two SET statements
+    pstmt = con.prepareStatement("SET search_path = 'public';SET search_path = 'public';"
+        + "DELETE FROM inttable WHERE a=?");
+    pstmt.setInt(1, 1);
+    count = pstmt.executeUpdate();
+    pstmt.close();
+    assertEquals("Expected update count to be 0", 0, count);
+  }
+
+  @Test
+  public void testCountForSelectStatementWithSet() throws SQLException {
+    PreparedStatement pstmt = con.prepareStatement("SET search_path = 'public';"
+        + "INSERT INTO inttable VALUES (?)");
+    for (int i = 0; i < 10; i++) {
+      pstmt.setInt(1, i);
+      pstmt.addBatch();
+    }
+    int[] counts = pstmt.executeBatch();
+    pstmt.close();
+    assertEquals("Expected update count to be 10", 10, counts.length);
+    //SELECT
+    pstmt = con.prepareStatement("SET search_path = 'public';"
+        + "SELECT * FROM inttable WHERE a>? AND a<?");
+    pstmt.setInt(1, 2);
+    pstmt.setInt(2, 8);
+    ResultSet rs = pstmt.executeQuery();
+    int rowCount = 0;
+    while (rs.next()) {
+      rowCount++;
+    }
+    rs.close();
+    pstmt.close();
+    assertEquals("Expected row count to be 5", 5, rowCount);
+    //With two SET statements
+    pstmt = con.prepareStatement("SET search_path = 'public';SET search_path = 'public';"
+        + "SELECT * FROM inttable WHERE a>? AND a<?");
+    pstmt.setInt(1, 2);
+    pstmt.setInt(2, 8);
+    rs = pstmt.executeQuery();
+    rowCount = 0;
+    while (rs.next()) {
+      rowCount++;
+    }
+    rs.close();
+    pstmt.close();
+    assertEquals("Expected row count to be 5", 5, rowCount);
```

Comment:
Does this belong to a different test method?

Reply:
Same test are improved